frontend/talentsearch/resources/js/components/experienceAccordion/ExperienceAccordion.tsx

```diff
@@ -0,0 +1,175 @@
+import React from "react";
+import { Accordion } from "@common/components/accordion/Accordion";
+import BriefCaseIcon from "@heroicons/react/solid/BriefcaseIcon";
+import {
+  AwardExperience,
+  CommunityExperience,
+  EducationExperience,
+  PersonalExperience,
+  WorkExperience,
+} from "../../api/generated";
+
+export interface AccordionProps {
+  anExperience: {
```

Comment:
This type in the props is pretty ugly and will be hard to maintain.  Would it be possible to make it the type `Experience`?  Or maybe `AwardExperience | CommunityExperience | EducationExperience | PersonalExperience | WorkExperience`?  And can we get away without the `experienceType` field and inspect the experience object directly to figure out what it is?  I saw [this page on type guards](https://medium.com/ovrsea/checking-the-type-of-an-object-in-typescript-the-type-guards-24d98d9119b0) that might give you some ideas on how to to do it though I've never tried this myself.  I know there's a `__typename` field that graphql makes that maybe you could use to differentiate, too.  I'm worried that might be a bad codesmell since it couples up your code with the API implementation so I maybe wouldn't try that unless absolutely necessary

Reply:
messy for sure
my approach was that I assumed there would be an object of unknown properties passed in with only experienceType field as a guaranteed input
I think I'll revisit making the input prettier later

Reply:
I just want to chime in here in support of peter's comment. With the way we've set up our project, when we're working on the frontend with data coming from the API, we want to avoid defining types manually. That's why we're using [codegen](https://www.graphql-code-generator.com/), which generates the TypeScript version of our graphql schema. If we always reference those generated types, we can ensure our frontend stays in sync with our API. Not to mention its less work than writing the types out manually ;)

Reply:
I suggest creating a new type somewhere, which is the union of the other Experience types. (This would be a superset of all of them, unlike `Experience` which only contains the common fields.) Not sure what to name it, since `Experience` is already taken 😆 . I have work with type guards before, I might be able to help when you get to that point.